gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/heap/MemHeapScanController.java

```diff
@@ -813,7 +813,7 @@ public final boolean next() throws StandardException {
                     + ", isIterOnPR=" + this.gfContainer.isPartitioned());
           }
           // a null owner in full scan will happen for remote entries
-          if (owner == null && !isGlobalScan) {
+          if (owner == null && !isGlobalScan && !Misc.getMemStore().isSnappyStore()) {
```

Comment:
This is for the ticket SNAP-652. There is a support for remote iterator already added in PartitionedRegion.java for snappystore case. But when the remote iterator is created, it's accompanying bucket-region is not available in the current vm (it's on the remote vm) and hence it is null. We would want to still be able to iterate over those entries.

The !isSnappyStore() ensures that we do not skip iterating over the (remote) entries from the remote node, even if its bucket is not available.